kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Watcher.java

```diff
@@ -19,6 +19,14 @@
 
 public interface Watcher<T> {
 
+  /**
+   * If the Watcher can reconnect itself after an error
+   * @return
+   */
+  default boolean reconnecting() {
```

Comment:
Is the reconnect-after-error behavior working for all Watcher managers?
I see in the issue that we're tackling this in the SharedInformer implementations, but we're probably missing this feature for the rest.
It might be safer to move the method out of the interface to the ReflectorWatcher implementation (in case we're unsure if the reconnect feature works everywhere).

Reply:
> Is the reconnect-after-error behavior working for all Watcher managers?

I only see isForcedClosed being referenced by WatcherWebSocketListener and WatchHTTPManager wrt reconnect behavior.  This change would have the same affect in both - they will still schedule reconnects after an exception.  But I do see an issue in that if it reaches the point where cannotReconnect returns false, it will hard loop - I'll make an update for that.

> It might be safer to move the method out of the interface to the ReflectorWatcher implementation (in case we're unsure if the reconnect feature works everywhere).

Do you mean doing an unwrap of the toggle and an instance of check for ReflectorWatcher in the AbstractWatchManager, rather than a reconnecting method on the Watcher interface?

Reply:
OK, sorry, I just saw this was being handled in the AbstractWatchManager, so it will be taken into account for any Manager implementation (unless it's not using the AbstractWatchManager ---> hopefully not).

Reply:
Just as a note, now there won't be a reconnection limit for those Watchers.

Reply:
> Just as a note, now there won't be a reconnection limit for those Watchers.

@manusa @rohanKanojia that is my understanding - an informer watch should always try to stay alive.  If it's possible for them to go down without user action, can we consider #3034 - so that it's easy to determine if the index state may be getting stale.

Reply:
+1 #3034 makes a lot of sense in the new context after this PR